libbeat/esleg/eslegclient/config.go

```diff
@@ -0,0 +1,78 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package eslegclient
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+)
+
+type config struct {
+	Hosts    []string          `config:"hosts" validate:"required"`
+	Protocol string            `config:"protocol"`
+	Path     string            `config:"path"`
+	Params   map[string]string `config:"parameters"`
+	Headers  map[string]string `config:"headers"`
+
+	TLS *tlscommon.Config `config:"ssl"`
+
+	ProxyURL     string `config:"proxy_url"`
+	ProxyDisable bool   `config:"proxy_disable"`
+
+	Username string `config:"username"`
+	Password string `config:"password"`
+	APIKey   string `config:"api_key"`
+
+	CompressionLevel int           `config:"compression_level" validate:"min=0, max=9"`
+	EscapeHTML       bool          `config:"escape_html"`
+	Timeout          time.Duration `config:"timeout"`
+}
```

Comment:
maybe as follow up, but these settings look quite generic, but HTTP focused. Just wondering if it would make sense to have these in libbeat/common/transport in one way or the other. We've got a many HTTP clients, with different set of settings.

Reply:
Yup, that makes sense to me. I'd prefer to do it as a follow up PR in which we look at HTTP settings across the various clients and consolidate the common ones somewhere under `libbeat/common/transport`.

Reply:
Created https://github.com/elastic/beats/issues/16856 to track this refactoring as a follow up.